File: 2023/day_one/main.ts
import * as fs from "fs";
import * as readline from "readline";

class DayOne {
  part_one(): number {
    const lineReader = readline.createInterface({
      input: fs.createReadStream("../inputs/day_one.txt"),
      terminal: false,
    });

    lineReader.on("line", (line) => {
      console.log(line);
    });
    return -1;
  }
}

let xdo = new DayOne();
xdo.part_one();

File: 2023/day_one/main.clj
(ns main
  (:gen-class))

(str "Hello" " " "World")

(defn PartOne []
  (def x [])
  (with-open [rdr (clojure.java.io/reader "../inputs/day_one.txt")]
    (def line (re-seq #"\d" (str (line-seq rdr))))
    (def digits (reduce conj [] [(first line) (last line)]))
    (conj x digits)
    (println x))
  x)


(println (PartOne))